src/cudadev/CondFormats/SiPixelROCsStatusAndMappingWrapper.cc

```diff
@@ -13,43 +13,40 @@
 #include "CUDACore/host_unique_ptr.h"
 #include "CUDADataFormats/gpuClusteringConstants.h"
 #include "CondFormats/SiPixelROCsStatusAndMappingWrapper.h"
+#include "CUDACore/copyAsync.h"
 
 SiPixelROCsStatusAndMappingWrapper::SiPixelROCsStatusAndMappingWrapper(SiPixelROCsStatusAndMapping const& cablingMap,
                                                                std::vector<unsigned char> modToUnp)
     : modToUnpDefault(modToUnp.size()), hasQuality_(true) {
-  cudaCheck(cudaMallocHost(&cablingMapHost, sizeof(SiPixelROCsStatusAndMapping)));
-  std::memcpy(cablingMapHost, &cablingMap, sizeof(SiPixelROCsStatusAndMapping));
-
+  // TODO: check if cudaStreamDefault is appropriate
+  auto cablingMapMetadata = cablingMap.soaMetadata();
+  cablingMapHostBuffer = cms::cuda::make_host_unique<std::byte[]>(cablingMapMetadata.byteSize(), cudaStreamDefault);
+  std::memcpy(cablingMapHostBuffer.get(), cablingMapMetadata.data(), cablingMapMetadata.byteSize());
   std::copy(modToUnp.begin(), modToUnp.end(), modToUnpDefault.begin());
 }
 
-SiPixelROCsStatusAndMappingWrapper::~SiPixelROCsStatusAndMappingWrapper() { cudaCheck(cudaFreeHost(cablingMapHost)); }
-
-const SiPixelROCsStatusAndMapping* SiPixelROCsStatusAndMappingWrapper::getGPUProductAsync(cudaStream_t cudaStream) const {
-  const auto& data = gpuData_.dataForCurrentDeviceAsync(cudaStream, [this](GPUData& data, cudaStream_t stream) {
-    // allocate
-    cudaCheck(cudaMalloc(&data.cablingMapDevice, sizeof(SiPixelROCsStatusAndMapping)));
-
-    // transfer
-    cudaCheck(cudaMemcpyAsync(
-        data.cablingMapDevice, this->cablingMapHost, sizeof(SiPixelROCsStatusAndMapping), cudaMemcpyDefault, stream));
-  });
+const SiPixelROCsStatusAndMapping & SiPixelROCsStatusAndMappingWrapper::getGPUProductAsync(cudaStream_t cudaStream) const {
+  const auto& data = gpuData_.dataForCurrentDeviceAsync(cudaStream, 
+    [this](GPUData& data, cudaStream_t stream) {
+      // allocate
+      data.allocate(pixelgpudetails::MAX_SIZE, stream);
+      // transfer
+      cms::cuda::copyAsync(data.cablingMapDeviceBuffer, this->cablingMapHostBuffer, 
+        data.cablingMapDevice.soaMetadata().byteSize(), stream);
+    }
+  );
   return data.cablingMapDevice;
 }
 
 const unsigned char* SiPixelROCsStatusAndMappingWrapper::getModToUnpAllAsync(cudaStream_t cudaStream) const {
   const auto& data =
       modToUnp_.dataForCurrentDeviceAsync(cudaStream, [this](ModulesToUnpack& data, cudaStream_t stream) {
-        cudaCheck(cudaMalloc((void**)&data.modToUnpDefault, pixelgpudetails::MAX_SIZE_BYTE_BOOL));
-        cudaCheck(cudaMemcpyAsync(data.modToUnpDefault,
+        data.modToUnpDefault = cms::cuda::make_device_unique<unsigned char []>(pixelgpudetails::MAX_SIZE_BYTE_BOOL, stream);
```

Comment:
Also here the use of `stream` for memory allocations is not, in principle, appropriate, but it probably doesn't matter in practice as the ES product outlives the event loop and all GPU work.

Reply:
Also leaving like this, then.